File: Stack Using LinkedList.c
#include<stdio.h>
#include<stdlib.h>

struct Node
{
	int data;
	struct Node *link;
};

void push(struct Node **top, int d)
{
	struct Node *r;
	r = (struct Node *)malloc(sizeof(struct Node));
	r->data = d;
	r->link = NULL;
	
	if(*top == NULL)
	{
		*top = r;
	}
	else
	{
		r->link = *top;
		*top = r;
	}
}

struct Node * pop(struct Node **top)
{
	struct Node *temp;
	temp = *top;
	*top = temp->link;
	printf("Deleted Data is: %i\n",temp->data);
	free(temp);
	return *top;
}

void display(struct Node *top)
{
	while(top != NULL)
	{
		printf("Data: %i\t Address: %x\t Top: %i\n",top->data,top->link,top);
		top = top->link;
	}
}

void main()
{
	struct Node *top = NULL;
	
	push(&top, 10);
	push(&top, 20);
	push(&top, 30);
	push(&top, 40);
	
	top = pop(&top);
	//top = pop(&top);
	//top = pop(&top);
		
	display(top);
}

File: StackUsingLinkedList.java
class Node 
{
    int data;
    Node link;
}

public class StackUsingLinkedList 
{
    public static void main(String args[]) 
    {
        Node top = new Node();
        top = null;
        top = Push(top , 10);
        top = Push(top , 20);
        top = Push(top , 30);
        top = Push(top , 40);
        top = Push(top , 50);

        top = Pop(top);
        //top = Pop(top);

        Display(top);
    }

    public static Node Push(Node top, int d)
    {
        Node r = new Node();
        r.data = d;
        r.link = null;
        if(top == null)
        {
            top = r;
        }
        else
        {
            r.link = top;
            top = r;
        }

        return top;
    }

    public static Node Pop(Node top)
    {
        if(top == null)
        {
            System.out.println("Stack UnderFlow"+"\n"+"Cannot Pop Elements");
        }
        else
        {
            Node temp = new Node();
            temp = top;
            top = temp.link;
            temp = null;
        }

        return top;
    }

    public static void Display(Node top)
    {
        if(top == null)
        {
            System.out.println("Stack UnderFlow");
        }
        else
        {
            while(top != null)
            {
                System.out.println("Data: "+top.data+"\t"+"Address: "+top.link+"\n");
                top = top.link;
            }
        }
    }
}